File: zazeutil/src/main/java/com/zaze/utils/ZCallback.java
package com.zaze.utils;

/**
 * Description : 回调接口（）
 *
 * @author : ZAZE
 * @version : 2017-01-06 - 11:16
 */
public abstract class ZCallback<D> {

    private String message;

    public void onError(int errorCode, String errorMsg) {
    }

    public D preNext(D d) {
        return d;
    }

    public abstract void onNext(D d);

    public abstract void onCompleted();

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: zazeutil/src/main/java/com/zaze/utils/ZCallback.kt
package com.zaze.utils

/**
 * Description : 回调接口（）
 *
 * @author : ZAZE
 * @version : 2017-01-06 - 11:16
 */
abstract class ZCallback<D> {

    var message: String? = null

    open fun onError(errorCode: Int, errorMsg: String?) {}

    open fun preNext(d: D?): D? {
        return d
    }

    abstract fun onNext(d: D?)

    abstract fun onCompleted()
}
